### PR TITLE
Convert latest signed root test to table.

### PR DESCRIPTION
**Pull request by Martin2112**
+ remove a couple of unused test structs.

### CODE REVIEW

File: server/log_rpc_server_test.go [hunk 1]
```diff
@@ -350,95 +348,90 @@ func TestQueueLeavesBeginFailsCausesError(t *testing.T) {
...
-		LogStorage:   mockStorage,
-	}
-	server := NewTrillianLogRPCServer(registry, fakeTimeSource)
-
-	_, err := server.GetLatestSignedLogRoot(context.Background(), &getLogRootRequest1)
-	if err == nil || !strings.Contains(err.Error(), "TX") {
-		t.Fatalf("Returned wrong error response when begin failed: %v", err)
-	}
+type latestRootTest struct {
+	req         trillian.GetLatestSignedLogRootRequest
```
Comment (daviddrysdale) [inline server/log_rpc_server_test.go]: `req` never varies, so in theory you could just drop it.

However, it looks like a get-root(invalid-tree-id) test gets dropped in the reshuffle so maybe keep `req` and reinstate a test case for that.

File: server/log_rpc_server_test.go [hunk 2]
```diff
@@ -350,95 +348,90 @@ func TestQueueLeavesBeginFailsCausesError(t *testing.T) {
...
-		t.Fatalf("Log root proto mismatch:\n%v\n%v", signedRoot1, resp.SignedLogRoot)
+		registry := extension.Registry{
+			AdminStorage: mockAdminStorage(ctrl, test.req.LogId),
+			LogStorage:   mockStorage,
+		}
+		s := NewTrillianLogRPCServer(registry, fakeTimeSource)
+		resp, err := s.GetLatestSignedLogRoot(context.Background(), &test.req)
+		if test.wantErr {
+			if err == nil {
+				t.Errorf("GetLatestSignedLogRoot(%+v)=nil, want: err", test.req)
```
Comment (daviddrysdale) [inline server/log_rpc_server_test.go]: Be nicer to indicate which result is involved: `"GetLatestSignedLogRoot()=_, nil; want _, err"` (passim)

File: server/log_rpc_server_test.go [hunk 3]
```diff
@@ -350,95 +348,90 @@ func TestQueueLeavesBeginFailsCausesError(t *testing.T) {
...
+			}
+		} else {
+			if err != nil {
+				t.Errorf("GetLatestSignedLogRoot(%+v)=%v, want: nil", test.req, err)
```
Comment (daviddrysdale) [inline server/log_rpc_server_test.go]: should probably `continue` here

File: server/log_rpc_server_test.go [hunk 4]
```diff
@@ -350,95 +348,90 @@ func TestQueueLeavesBeginFailsCausesError(t *testing.T) {
...
+			}
+			// Ensure we got the expected root back.
+			if got, want := resp, &test.wantRoot; !proto.Equal(got, want) {
```
Comment (daviddrysdale) [inline server/log_rpc_server_test.go]: rename `resp` to `got` and you could drop one of the scope-locals here; if it were me, I'd drop the other one too as `test.wantRoot` already has the magic "want" substring

File: server/log_rpc_server_test.go (hunk 1 above)
Comment (Martin2112) [reply server/log_rpc_server_test.go]: It looked like it wasn't testing anything useful but having read it again I think it's testing an admin store failure, which can probably still use the same req.

Comment (Martin2112) [reply server/log_rpc_server_test.go]: I've checked this out and the mock admin storage is never called so the removed test was not adding value. The admin storage would be called from storage, but it's also a mock and we bypass all that stuff.

File: server/log_rpc_server_test.go (hunk 2 above)
Comment (Martin2112) [reply server/log_rpc_server_test.go]: Done.

File: server/log_rpc_server_test.go (hunk 3 above)
Comment (Martin2112) [reply server/log_rpc_server_test.go]: Done.

File: server/log_rpc_server_test.go (hunk 4 above)
Comment (Martin2112) [reply server/log_rpc_server_test.go]: Done.

**Comment by Martin2112** [comment]
PTAL.

File: server/log_rpc_server_test.go [hunk 5]
```diff
@@ -1278,20 +1285,21 @@ func TestGetConsistencyProof(t *testing.T) {
 		response, err := server.GetConsistencyProof(context.Background(), &test.req)
 
 		if test.wantErr {
-			if err == nil {
-				t.Errorf("GetConsistencyProof(%+v)=nil, want: err", test.req)
+			if err == nil || !strings.Contains(err.Error(), test.errStr) {
+				t.Errorf("GetConsistencyProof(%+v)=_,nil, want: _,err contains: %s but got: %v", test.req, test.errStr, err)
```
Comment (daviddrysdale) [inline server/log_rpc_server_test.go]: the error message is now inaccurate -- something like:
```
t.Errorf("GetConsistencyProof(%+v)=_, %v; want _, err containing %q", test.req, err, test.errStr)
```
would cover both cases

File: server/log_rpc_server_test.go [hunk 6]
```diff
@@ -1135,6 +1133,7 @@ func TestGetSequencedLeafCount(t *testing.T) {
 type consistProofTest struct {
 	req         trillian.GetConsistencyProofRequest
 	wantErr     bool
+	errStr      string
```
Comment (daviddrysdale) [inline server/log_rpc_server_test.go]: Could just use `errStr!=""` to indicate `wantErr`

File: server/log_rpc_server_test.go [hunk 7]
```diff
@@ -1278,20 +1285,21 @@ func TestGetConsistencyProof(t *testing.T) {
...
 		if test.wantErr {
-			if err == nil {
-				t.Errorf("GetConsistencyProof(%+v)=nil, want: err", test.req)
+			if err == nil || !strings.Contains(err.Error(), test.errStr) {
+				t.Errorf("GetConsistencyProof(%+v)=_,nil, want: _,err contains: %s but got: %v", test.req, test.errStr, err)
 			}
 		} else {
 			if err != nil {
-				t.Errorf("GetConsistencyProof(%+v)=%v, want: nil", test.req, err)
+				t.Errorf("GetConsistencyProof(%+v)=%v,err, want: _,nil", test.req, err)
```
Comment (daviddrysdale) [inline server/log_rpc_server_test.go]: `%v,err` => `_, %v`

File: server/log_rpc_server_test.go (hunk 6 above)
Comment (Martin2112) [reply server/log_rpc_server_test.go]: Done.

File: server/log_rpc_server_test.go (hunk 5 above)
Comment (Martin2112) [reply server/log_rpc_server_test.go]: Done.

File: server/log_rpc_server_test.go (hunk 7 above)
Comment (Martin2112) [reply server/log_rpc_server_test.go]: Done.

**Comment by Martin2112** [comment]
Is this OK now?

**Comment by daviddrysdale** [comment]
Sorry, didn't notice the update.

**Pull request merged**